chia/simulator/socket.py

```diff
@@ -7,17 +7,28 @@
 recent_ports: Set[int] = set()
 
 
-def find_available_listen_port(name: str = "free") -> int:
+def find_available_listen_port(name: str = "free", ipv6_required: bool = False) -> int:
     global recent_ports
 
     while True:
         port = secrets.randbelow(0xFFFF - 1024) + 1024
         if port in recent_ports:
             continue
 
-        with socket.socket() as s:
-            try:
+        try:
+            with socket.socket(type=socket.SOCK_STREAM) as s:  # TCP
+                s.bind(("127.0.0.1", port))
+            with socket.socket(type=socket.SOCK_DGRAM) as s:  # UDP
                 s.bind(("127.0.0.1", port))
```

Comment:
We're forcing all existing uses to open UDP ports as well?

Reply:
yeah, i can make that separate.